pyfr/solvers/baseadvec/elements.py

```diff
@@ -99,3 +97,51 @@ def filter_soln(uin):
                 return self._be.ordered_meta_kernel([mul, copy])
 
             kernels['filter_soln'] = filter_soln
+
+        
+        shock_capturing = self.cfg.get('solver', 'shock-capturing', 'none')
+        if shock_capturing == 'entropy-filter':
+            nonce = args[2]
```

Comment:
Fragile. If you do need the nonce declare the full argument list in set_backend.